sdk/metrics/src/main/java/io/opentelemetry/sdk/metrics/aggregator/DoubleMinMaxSumCountAggregator.java

```diff
@@ -11,77 +11,85 @@
 import javax.annotation.concurrent.ThreadSafe;
 
 @ThreadSafe
-public final class DoubleMinMaxSumCountAggregator extends Aggregator<MinMaxSumCountAccumulation> {
-  private static final AggregatorFactory<MinMaxSumCountAccumulation> AGGREGATOR_FACTORY =
-      new AggregatorFactory<MinMaxSumCountAccumulation>() {
-        @Override
-        public Aggregator<MinMaxSumCountAccumulation> getAggregator() {
-          return new DoubleMinMaxSumCountAggregator();
-        }
+public final class DoubleMinMaxSumCountAggregator
+    implements Aggregator<MinMaxSumCountAccumulation> {
+  private static final DoubleMinMaxSumCountAggregator INSTANCE =
+      new DoubleMinMaxSumCountAggregator();
 
-        @Override
-        public MinMaxSumCountAccumulation accumulateDouble(double value) {
-          return MinMaxSumCountAccumulation.create(1, value, value, value);
-        }
-      };
-
-  private final ReentrantReadWriteLock lock = new ReentrantReadWriteLock();
-  // The current value. This controls its own internal thread-safety via method access. Don't
-  // try to use its fields directly.
-  @GuardedBy("lock")
-  private final DoubleState current = new DoubleState();
-
-  public static AggregatorFactory<MinMaxSumCountAccumulation> getFactory() {
-    return AGGREGATOR_FACTORY;
+  /**
+   * Returns the instance of this {@link Aggregator}.
+   *
+   * @return the instance of this {@link Aggregator}.
+   */
+  public static Aggregator<MinMaxSumCountAccumulation> getInstance() {
+    return INSTANCE;
   }
 
   private DoubleMinMaxSumCountAggregator() {}
 
   @Override
-  protected MinMaxSumCountAccumulation doAccumulateThenReset() {
-    lock.writeLock().lock();
-    try {
-      MinMaxSumCountAccumulation toReturn =
-          MinMaxSumCountAccumulation.create(current.count, current.sum, current.min, current.max);
-      current.reset();
-      return toReturn;
-    } finally {
-      lock.writeLock().unlock();
-    }
+  public AggregatorHandle<MinMaxSumCountAccumulation> createHandle() {
+    return new Handle();
   }
 
   @Override
-  protected void doRecordDouble(double value) {
-    lock.writeLock().lock();
-    try {
-      current.record(value);
-    } finally {
-      lock.writeLock().unlock();
-    }
+  public MinMaxSumCountAccumulation accumulateDouble(double value) {
+    return MinMaxSumCountAccumulation.create(1, value, value, value);
   }
 
-  private static final class DoubleState {
-    private long count;
-    private double sum;
-    private double min;
-    private double max;
+  static final class Handle extends AggregatorHandle<MinMaxSumCountAccumulation> {
```

Comment:
If we're ok with not testing the exact type returned by the createHandle method, we could make these handle implementations private.

Reply:
I had them initially private, then when I had to fix the tests I did this. We should follow up on this, for the moment I am keeping this way to not remove tests.